src/shared/tests.h

```diff
@@ -200,3 +200,35 @@ static inline int run_test_table(void) {
         DEFINE_TEST_MAIN_FULL(log_level, intro, NULL)
 #define DEFINE_TEST_MAIN(log_level)                     \
         DEFINE_TEST_MAIN_FULL(log_level, NULL, NULL)
+
+#define ASSERT_OK(expr)                                                                  \
+    do {                                                                                 \
+        if (!(expr)) {                                                                   \
+            sd_journal_print(LOG_ERR, "Assertion failed: %s", #expr);                    \
+            abort();                                                                     \
+        }                                                                                \
+    } while (false)
```

Comment:
`ASSERT_OK` is for checking if the `expr` is an error. it should check if the given expression is negative (all errors in our codebase are negative values). The actual error should also be logged using `log_error_errno`

Reply:
Thanks for the review.